File: lab10/reservationModel/src/main/java/model/Reservation.java
package model;

import model.HasID;
import model.Journey;

import java.io.Serializable;

public class Reservation implements HasID<String>, Serializable {
    private String id;
    private String customerName;
    private String phoneNumber;
    private int noTickets;
    private Journey journey;

    public Reservation(String id, String customerName, String phoneNumber, int noTickets, Journey journey) {
        this.id = id;
        this.customerName = customerName;
        this.phoneNumber = phoneNumber;
        this.noTickets = noTickets;
        this.journey = journey;
    }

    @Override
    public String getID() {
        return id;
    }

    @Override
    public void setID(String newID) {
        id= newID;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public int getNoTickets() {
        return noTickets;
    }

    public void setNoTickets(int noTickets) {
        this.noTickets = noTickets;
    }

    public Journey getJourney() {
        return journey;
    }

    public void setJourney(Journey journey) {
        this.journey = journey;
    }
}

File: lab10_C#/ReservationGrpc/model/Reservation.cs
using System;
using System.Runtime.CompilerServices;
using System.Security.Policy;

namespace Reservations.model
{
    [Serializable]

    public class Reservation : IHasID<string>
    {
        private string id;
        private string customerName;
        private string phoneNumber;
        private int noTickets;
        private Journey journey;

        public Reservation(string id, string customerName, string phoneNumber, int noTickets, Journey journey)
        {
            this.id = id;
            this.customerName = customerName;
            this.phoneNumber = phoneNumber;
            this.noTickets = noTickets;
            this.journey = journey;
        }

        public string ID
        {
            get { return id; }
            set { id = value; }
        }


        public string CustomerName
        {
            get => customerName;
            set => customerName = value;
        }

        public string PhoneNumber
        {
            get => phoneNumber;
            set => phoneNumber = value;
        }

        public int NoTickets
        {
            get => noTickets;
            set => noTickets = value;
        }

        public Journey Journey
        {
            get => journey;
            set => journey = value;
        }


        public override string ToString()
        {
            return string.Format("[Reservations: Id={0}, customerName={1}, phoneNumber={2}, noTickets={3}, journey={4}",
                id, customerName, phoneNumber, journey);
        }
    }
}